### PR TITLE
Add RegExp methods

### PR DESCRIPTION
**Pull request by petele**
There are a handful of deprecated methods that I've included, but think we should probably remove? Looking for confirmation please.

* `compile`
* `input`
* `lastMatch`
* `lastParent`
* `leftContext`
* `rightContext`

### CODE REVIEW

File: features/regexp.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Regular Expressions (RegExp)
+description: The RegExp object is used for matching text with a regular expression.
+spec: https://tc39.es/ecma262/multipage/text-processing.html#sec-regexp-regular-expression-objects
+status:
+  compute_from: javascript.builtins.RegExp
+compat_features:
+  - javascript.builtins.RegExp
+  - javascript.builtins.RegExp.RegExp
+  - javascript.builtins.RegExp.compile
```
Comment (petele) [inline features/regexp.yml:9]: this method is deprecated - should we drop it from this list?

File: features/regexp.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - javascript.builtins.RegExp.dotAll
+  - javascript.builtins.RegExp.exec
+  - javascript.builtins.RegExp.flags
+  - javascript.builtins.RegExp.global
+  - javascript.builtins.RegExp.global.prototype_accessor
+  - javascript.builtins.RegExp.hasIndices
+  - javascript.builtins.RegExp.ignoreCase
+  - javascript.builtins.RegExp.ignoreCase.prototype_accessor
+  - javascript.builtins.RegExp.input
```
Comment (petele) [inline features/regexp.yml:18]: this method is deprecated - should we drop it from this list?

File: features/regexp.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: Regular Expressions (RegExp)
+description: The RegExp object is used for matching text with a regular expression.
```
Comment (ddbeck) [inline features/regexp.yml:2]: Feature names are sentence case and also-known-as text goes in the description (if needed, though not in this case)

```suggestion
name: Regular expressions
description: The `RegExp` object represents a regular expression, a notation for matching text patterns.
```

File: features/regexp.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  - javascript.builtins.RegExp.toString
+  - javascript.builtins.RegExp.toString.escaping
+  - javascript.builtins.RegExp.unicode
+  - javascript.builtins.RegExp.unicodeSets
+  - javascript.builtins.RegExp.@@match
+  - javascript.builtins.RegExp.@@matchAll
+  - javascript.builtins.RegExp.@@replace
+  - javascript.builtins.RegExp.@@search
+  - javascript.builtins.RegExp.@@species
+  - javascript.builtins.RegExp.@@split
```
Comment (ddbeck) [inline features/regexp.yml:44]: We also need `javascript.grammar.regular_expression_literals` and all of the non-deprecated `javascript.regular_expressions.*` keys. It's possible there are some distinct features in there, but it'll be easier to pick them out with the dist file.

**Review by ddbeck — CHANGES_REQUESTED** [review]

**Comment by petele** [comment]
Added additional keys, and removed all deprecated functions.

**Review by ddbeck — APPROVED** [review]
Thank you, Pete!

**Pull request merged**